firebase-firestore/ktx/src/main/kotlin/com/google/firebase/firestore/ktx/Firestore.kt

```diff
@@ -162,3 +172,71 @@ class FirebaseFirestoreKtxRegistrar : ComponentRegistrar {
     override fun getComponents(): List<Component<*>> =
             listOf(LibraryVersionComponent.create(LIBRARY_NAME, BuildConfig.VERSION_NAME))
 }
+
+/**
+ * Transforms a [DocumentReference] into a coroutine [Flow]
+ *
+ * **Backpressure handling**: by default this method conflates items. If the consumer isn't fast enough,
+ * it might miss some values but is always guaranteed to get the latest value. Use [bufferCapacity]
+ * to change that behaviour
+ *
+ * @param metadataChanges controls metadata-only changes. Default: [MetadataChanges.EXCLUDE]
+ * @param bufferCapacity the buffer capacity as in [Flow.buffer] or null to not buffer at all
+ */
+fun DocumentReference.toFlow(
```

Comment:
We are considering calling this `DocumentReference.documentSnapshot` instead, to better describe the return type:

```kotlin
docRef.documentSnapshot.collect { snapshot ->
    // handle returned document
}
```

I would like to hear y'all thoughts.

Reply:
Not sure what the best practices are. It'd be worth investigating what other libs are doing. In all cases, we need to account for `metadataChanges` and `bufferCapacity` parameters so it can't really be a `val`. Maybe this instead?

```kotlin
docRef.snapshots().collect { snapshot ->
  // do something with snapshot
}
```

And if customization is required, this becomes:

```kotlin
docRef.snapshots(metadataChanges, bufferCapacity).collect { snapshot ->
  // do something with snapshot
}
```

Reply:
I'm with @martinbonnin here. It should be `snapshots()` or `documentSnapshots()`. Plural, because a `Flow` usually returns multiple values. The `document` in `documentSnapshots()` is redundant because it should be clear that a `DocumentReference` returns `DocumentSnapshot`. So `snapshots()` is my favorite candidate.

Reply:
I agree it should be plural. 👍 
Sorry for the `val` confusion, that was a mistake on my side, I forgot to add parenthesis.